src/ra_errors.h

```diff
@@ -0,0 +1,42 @@
+#ifndef RA_ERRORS_H
+#define RA_ERRORS_H
+#pragma once
+
+// TODO: Make a pch for repetitive includes
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>    
+#include <system_error>
+
+namespace ra {
+
+using tstring = std::basic_string<TCHAR>;
+
+_EXTERN_C
+
+_Success_(return != 0)
+int CALLBACK ShowError(_In_ const tstring& message, _In_ HWND hwnd = ::GetActiveWindow()) noexcept;
+_NODISCARD bool CALLBACK InvalidModule() noexcept;
+
+// Shows the last system error with a message without throwing
+int CALLBACK ShowLastError() noexcept;
+_END_EXTERN_C
```

Comment:
Some of these aren't marked with `[[nodiscard]]` because I felt it wasn't necessary to absolutely use them.
Also a lot of these functions I put use `__stdcall `mostly because of calls to the Windows API and it's similar to `__thiscall `(calling convention for classes)